test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs

```diff
@@ -0,0 +1,336 @@
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <packageSource key=""source1"">
+            <namespace id=""NuGet.*"" />
+        </packageSource>
+        <packageSource key=""source2"">
+            <namespace id=""NuGet.Common.*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.AreNamespacesEnabled);
+            configuration.Namespaces.Should().HaveCount(2);
+
+            var configuredSources = configuration.GetConfiguredPackageSources("NuGet.Common1");
+            Assert.Equal(1, configuredSources.Count);
+            Assert.Equal("source1", configuredSources.First());
+
+            configuredSources = configuration.GetConfiguredPackageSources("NuGet.Common");
+            Assert.Equal(1, configuredSources.Count);
+            Assert.Equal("source1", configuredSources.First());
+
+            configuredSources = configuration.GetConfiguredPackageSources("NuGet.Common.Package.Id");
+            Assert.Equal(1, configuredSources.Count);
+            Assert.Equal("source2", configuredSources.First());
+        }
+
+        [Fact]
+        public void SearchTree_InvalidInput_Throws()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <packageSource key=""source1"">
+            <namespace id=""NuGet.*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+
+            var exception = Assert.Throws<ArgumentException>(
+                () => configuration.GetConfiguredPackageSources(null));
+
+            Assert.Equal("Argument cannot be null, empty, or white space only." + Environment.NewLine + "Parameter name: term", exception.Message);
+
+            exception = Assert.Throws<ArgumentException>(
+                () => configuration.GetConfiguredPackageSources(string.Empty));
+
+            Assert.Equal("Argument cannot be null, empty, or white space only." + Environment.NewLine + "Parameter name: term", exception.Message);
+
+            exception = Assert.Throws<ArgumentException>(
+                () => configuration.GetConfiguredPackageSources(" "));
+
```

Comment:
I would use `TheoryAttribute` to provide different invalid inputs for the testcase instead of copying code for every case.

Reply:
I explained several times why it's not working here.